cpp/src/Node.cpp

```diff
@@ -498,6 +501,42 @@ void Node::AdvanceQueries
 				}
 				break;
 			}
+			case QueryStage_Groups:
```

Comment:
Yes if the number of groups is not yet retrieved QueryStage_Association would also retrieve this number of groups. 
Groups != Associations. Groups hold associations. A group can hold a number of associations. 

QueryStage_Association retrieves the actual associations and updates the routes.  
Here we also add the controller as association based on config options(auto=true). This could be enhanced to also take into account the group profile that is retrieved by the AGI class. 
(the group profile is not yet retrieved)

Thats why I have separate states for the number of groups, group info and associations.
Putting it all in  QueryStage_Static would not enforce the needed sequence.